### PR TITLE
Fix issue 4710 by updating schedule

### LINKED ISSUE #4710: circuit -> schedule raises exception
### Information

- **Qiskit Terra version**: master
- **Python version**:
- **Operating system**:

### What is the current behavior?
```python
ghz = QuantumCircuit(5, 5)
ghz.h(0)
ghz.cx(range(4), range(1,5))
ghz.barrier()
ghz.measure(range(5), range(5))

sch = schedule(ghz, backend)
```

gives:

AttributeError: 'NoneType' object has no attribute 'instruction_schedule_map'

This works on older versions.

### PR DESCRIPTION
**Pull request by Chriscrosser3310**
### Summary

Fix #4710 

### Details and comments

backend.dafults() returns None if the backend does not support pulse, and calling schedule(...) on the backend gives a confusing error message.
The solution I made is raising a QiskitError if backend.defaults() is None.

### CODE REVIEW

File: qiskit/compiler/schedule.py [hunk 1]
```diff
@@ -65,6 +65,8 @@ def schedule(circuits: Union[QuantumCircuit, List[QuantumCircuit]],
         if backend is None:
             raise QiskitError("Must supply either a backend or InstructionScheduleMap for "
                               "scheduling passes.")
+        if backend.defaults() is None:
+            raise QiskitError("The backend does not support pulse.")
         inst_map = backend.defaults().instruction_schedule_map
```
Comment (taalexander) [inline qiskit/compiler/schedule.py:70]:
```suggestion
        defaults = backend.defaults()
        if defaults is None:
            raise QiskitError("The backend defaults are unavailable. The backend may not support pulse.")
        inst_map = defaults.instruction_schedule_map
```

**Comment by Chriscrosser3310** [comment]
@taalexander Thanks for the feedback! I wonder what kinds of tests should I add?

**Comment by taalexander** [comment]
Please try using a fake backend like https://github.com/Qiskit/qiskit-terra/blob/99f1b41fa54733bb972ba9d961afdf8646b8a4dd/qiskit/test/mock/fake_1q.py#L24 and verify that is `assertRaises` when defaults are unavailable.

**Comment by Chriscrosser3310** [comment]
@taalexander Thanks for the guide! I've updated the pull request. It seems none of the fake backends will return None for .defaults(), so I created a FakeBackend object and bound its .default() to .configuration()

**Comment by Chriscrosser3310** [comment]
@taalexander Thanks so much! Not sure if this is a good place to ask, but generally what happens after the pull request is approved?

**Comment by taalexander** [comment]
I've labelled with `automerge`. This signals to our merge bot to merge your code in when it has passed all CI tests. Unfortunately our CI pipeline has been having some troubles due to a recent dependency release. I've manually restarted checks.

**Pull request merged**